data_labeling_examples/src/main/java/com/oracle/datalabelingservicesamples/scripts/SingleLabelDatasetBulkLabelingScript.java

```diff
@@ -0,0 +1,150 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import com.oracle.bmc.datalabelingservicedataplane.model.CreateAnnotationDetails;
+import com.oracle.bmc.datalabelingservicedataplane.model.Dataset;
+import com.oracle.bmc.datalabelingservicedataplane.model.Entity;
+import com.oracle.bmc.datalabelingservicedataplane.model.GenericEntity;
+import com.oracle.bmc.datalabelingservicedataplane.model.Label;
+import com.oracle.bmc.datalabelingservicedataplane.model.LabelName;
+import com.oracle.bmc.datalabelingservicedataplane.model.Record.LifecycleState;
+import com.oracle.bmc.datalabelingservicedataplane.model.RecordSummary;
+import com.oracle.bmc.datalabelingservicedataplane.requests.CreateAnnotationRequest;
+import com.oracle.bmc.datalabelingservicedataplane.requests.GetDatasetRequest;
+import com.oracle.bmc.datalabelingservicedataplane.requests.ListRecordsRequest;
+import com.oracle.bmc.datalabelingservicedataplane.responses.GetDatasetResponse;
+import com.oracle.bmc.datalabelingservicedataplane.responses.ListRecordsResponse;
+import com.oracle.datalabelingservicesamples.requests.Config;
+
+import lombok.extern.slf4j.Slf4j;
+
+/*
+ * 
+ * This Script takes following input for Bulk Labeling:
+ * 
+ * DATASET_ID : the id of the dataset that you want to bulk label
+ * LABELS : List of labels against which the record names will be matched
+ * LABELING_ALGORITHM : The pattern matching algorithm that will determine the label of any record. 
+ * 						Currently following algorithms are supported: FIRST_LETTER and FIRST_MATCH
+ * FIRST_MATCH_REGEX_PATTERN : When the labeling algorithm is FIRST_MATCH, then we match against this pattern
+ * 
+ */
+
+@Slf4j
+public class SingleLabelDatasetBulkLabelingScript {
+
+	static ExecutorService executorService;
+	static Dataset dataset;
+	static List<String> successRecordIds = Collections.synchronizedList(new ArrayList<String>());
+	static List<String> failedRecordIds = Collections.synchronizedList(new ArrayList<String>());
+
+	static {
+		executorService = Executors.newFixedThreadPool(Config.INSTANCE.getThreadCount());
+	}
+
+	public static void main(String[] args) throws InterruptedException, ExecutionException {
+		long startTime = System.nanoTime();
+		String datasetId = Config.INSTANCE.getDatasetId();
+		validateRequest(datasetId);
+
+		log.info("Starting Bulk Labeling for dataset: {}", dataset.getDisplayName());
+		Boolean isUnlabeledRecordAvailable = true;
+		int recordCount = 0;
+
+		while (isUnlabeledRecordAvailable) {
+			ListRecordsRequest listRecordsRequest = ListRecordsRequest.builder().datasetId(datasetId)
+					.compartmentId(dataset.getCompartmentId()).lifecycleState(LifecycleState.Active).isLabeled(false)
+					.limit(1000).build();
```

Comment:
Probably move the value 1000 to a constants file?

Reply:
Done